common/components/common/projects/EditProjectForm.jsx

```diff
@@ -0,0 +1,188 @@
+      this.setState({
+        formFields: {
+          project_name: project.project_name,
+          project_location: project.project_location,
+          project_url: project.project_url,
+          project_description: project.project_description,
+          project_issue_area: project.project_issue_area && project.project_issue_area[0],
+          project_links: project.project_links,
+          project_files: project.project_files,
+          project_thumbnail: project.project_thumbnail
+        }
+      });
+      this.checkFormValidity();
+    }
+  }
+  
+  handleLoadProjectError(error: APIError): void {
+    this.setState({
+      error: "Failed to load project information"
+    });
+  }
+  
+  onFormFieldChange(formFieldName: string, event: SyntheticInputEvent<HTMLInputElement>): void {
+    this.state.formFields[formFieldName] = event.target.value;
+    this.forceUpdate();
+    this.checkFormValidity();
+  }
+  
+  onComponentChange(formFieldName: string, newValue: string): void {
+    this.state.formFields[formFieldName] = newValue;
+    this.checkFormValidity();
+  }
+  
+  checkFormValidity(): void {
+    var formFields = this.state.formFields;
+    
+    var valid = !_.isEmpty(formFields["project_name"]) && !_.isEmpty(formFields["project_description"]);
+    this.setState({
+      formIsValid: valid
+    });
+  }
+  
+  render(): React$Node {
+    if(this.state.error) {
+      return this._renderError();
+    } else {
+      return this._renderForm();
+    }
+  }
+  
+  _renderError(): React$Node {
+    return (
+      <div className="EditProjectForm-error">
+        {this.state.error}
+      </div>
+    );
+  }
+  
+  _renderForm(): React$Node {
+    return (
+      <div className="EditProjectForm-root">
+        
+        <DjangoCSRFToken/>
+        
+        <div className="form-group">
+          <ImageUploadFormElement form_id="project_thumbnail_location" currentImage={this.state.formFields.project_thumbnail}/>
+        </div>
+        
+        <h2 className="form-group subheader">DETAILS</h2>
+        <div className="form-group">
+          <label htmlFor="project_name">Project Name</label>
+          <input type="text" className="form-control" id="project_name" name="project_name" maxLength="200"
+                 value={this.state.formFields.project_name} onChange={this.onFormFieldChange.bind(this, "project_name")}/>
+        </div>
+        
+        <div className="form-group">
+          <label htmlFor="project_location">Project Location</label>
+          <input type="text" className="form-control" id="project_location" name="project_location" maxLength="200"
+                 value={this.state.formFields.project_location} onChange={this.onFormFieldChange.bind(this, "project_location")}/>
+        </div>
+        <div className="form-group">
+          <label htmlFor="project_url">Website URL</label>
+          <input type="text" className="form-control" id="project_url" name="project_url" maxLength="200"
```

Comment:
what's this className do? Also why the low maxLength value?

Reply:
form-control is another bootstrap styling.

And yeah, that is a low maxLength value for a url field.  Will change it to be 2000 in line with our other urls.